yt/utilities/on_demand_imports.py

```diff
@@ -56,6 +56,17 @@ class netCDF4_imports:
     _name = "netCDF4"
     _Dataset = None
 
+    def __init__(self):
+        # this ensures the import ordering between netcdf4 and h5py. If hy5py is
+        # imported first, can get file lock errors on some systems (including travis-ci)
+        # so we need to do this before initializing h5py_imports()!
+        # similar to this issue https://github.com/pydata/xarray/issues/2560
+        try:
+            import netCDF4  # noqa F401
+        except ImportError:
+            pass
+        super(netCDF4_imports, self).__init__()
```

Comment:
I don't get this. This class isn't inheriting from anywhere, so what's the effect of super here ?

Reply:
That might be my fault. I just blindly copied `h5py_imports.__init__` without reading it.

Reply:
a good question. The issue we were seeing was due to the import order of `h5py` and `netCDF4` (if `h5py` is imported first, using `netCDF4` to read files fails), we were not able to figure out why exactly. @Xarthisius figured out this workaround, he might have some insight.

Reply:
I think @neutrinoceros only meant that `super()` is not necessary here, cause there's no parent `__init__`

Reply:
We did try the environment variable solution mentioned in the comment [link](https://github.com/pydata/xarray/issues/2560), that didn't work. Didn't seem to be version related and couldn't reproduce it locally, only in the test environment.

Reply:
> I think @neutrinoceros only meant that super() is not necessary here, cause there's no parent __init__

exactly

Reply:
Ok, got it. We'll take out the super then

Reply:
removed the super call in the latest commit